sdk/spring-sdk/src/main/scala/kalix/springsdk/KalixConfiguration.scala

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2021 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kalix.springsdk
+
+import kalix.javasdk.action.ActionCreationContext
+import kalix.javasdk.valueentity.ValueEntityContext
+import kalix.javasdk.view.ViewCreationContext
+import kalix.springsdk.impl.KalixServer
+import kalix.springsdk.impl.KalixServer.ActionCreationContextFactoryBean
+import kalix.springsdk.impl.KalixServer.ValueEntityContextFactoryBean
+import kalix.springsdk.impl.KalixServer.ViewCreationContextFactoryBean
+import org.springframework.context.ApplicationContext
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+
+@Configuration
+class KalixConfiguration(applicationContext: ApplicationContext) {
+
+  private val threadLocalActionContext = new ThreadLocal[ActionCreationContext]
+  private val threadLocalValueEntityContext = new ThreadLocal[ValueEntityContext]
+  private val threadLocalViewContext = new ThreadLocal[ViewCreationContext]
+
+  @Bean
+  def kalixServer: KalixServer =
+    new KalixServer(applicationContext, threadLocalActionContext, threadLocalValueEntityContext, threadLocalViewContext)
+
+  // the FactoryBeans below will allow Spring to find an instance of the respective context
+  // whenever it needs to instantiate a Kalix component requiring a context
+  @Bean
+  def actionCreationContextFactoryBean: ActionCreationContextFactoryBean =
+    new ActionCreationContextFactoryBean(threadLocalActionContext)
+
+  @Bean
+  def valueEntityContext: ValueEntityContextFactoryBean =
+    new ValueEntityContextFactoryBean(threadLocalValueEntityContext)
+
+  @Bean
+  def viewCreationContext: ViewCreationContextFactoryBean =
+    new ViewCreationContextFactoryBean(threadLocalViewContext)
```

Comment:
None of those should be available to users, but only accessible internally. If possible, we should block access to them.

Reply:
hmm, I think we are fine here. They won't be available in the application context that users will have access.